pkgs/os-specific/linux/nixos-rebuild/nixos-rebuild.sh

```diff
@@ -214,7 +214,12 @@ nixBuild() {
 }
 
 nixFlakeBuild() {
-    if [ -z "$buildHost" ]; then
+    if [[ -z "$buildHost" && -z "$targetHost" ]] &&
+       ! [ "$action" = switch -o "$action" = boot ]
```

Comment:
Should this set of actions also include `test` which changes current system but doesn't update next boot as `switch` do?

Or maybe `./result` should be produced only from `build*` actions. At least man-page in 21.05 indicates symlinks being produced for those and silent about others.